---
title: "Example 2: RSS-BVSR with Various LD Matrices"
author: Xiang Zhu
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

[Zhu and Stephens (2017)]: https://projecteuclid.org/euclid.aoas/1507168840
[`example2.m`]: https://github.com/stephenslab/rss/blob/master/examples/example2.m

## Overview

This example illustrates the impact of different LD estimates on the
RSS results. Three types of estimated LD matrices are considered:
cohort sample LD, panel sample LD and shrinkage panel sample LD in
[Wen and Stephens, (2010)](https://www.ncbi.nlm.nih.gov/pubmed/21479081)
This example is closely related to Section 4.1 of [Zhu and Stephens (2017)][]. 

The single-SNP summary-level data are computed from a simulated GWAS dataset.
The simulation scheme is described in Section 4.1 [Zhu and Stephens (2017)][].
Specifically, 10 "causal" SNPs are randomly drawn from 982 SNPs on chromosome 16,
with effect sizes coming from standard normal ${\cal N}(0,1)$.
Effect sizes of remaining SNPs are zero. The true PVE (SNP heritability) is 0.2.

Three types of LD estimates are considered here.

- cohort sample LD: the sample correlation matrix using genotypes in the cohort (WTCCC UK Blood Service Control Group)

- shrinkage panel sample LD: the shrinkage correlation matrix
([Wen and Stephens, 2010](https://www.ncbi.nlm.nih.gov/pubmed/21479081))
using genotypes in the panel (WTCCC 1958 British Birth Cohort) 

- panel sample LD: the sample correlation matrix using genotypes in the panel (WTCCC 1958 British Birth Cohort) 

To reproduce results of Example 2, please read the step-by-step guide below and run [`example2.m`][].
Before running [`example2.m`][], please first install the
[MCMC subroutines](https://github.com/stephenslab/rss/tree/master/src).
Please find installation instructions [here](setup_mcmc.html).

## Step-by-step illustration

**Step 1**. Download data files.

All data files required to run this example are freely available at Zenodo
[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.12636406.svg)](https://doi.org/10.5281/zenodo.12636406).
Please contact me if you have trouble accessing this file.
After a complete download, you should see the following files.

```{r, eval=FALSE, engine='zsh'}
$ tree ./
./
├── example2.mat
└── readme

0 directories, 2 files
```

The data file `example2.mat` contains the following elements.

- `betahat`: 982 by 1 vector, single-SNP effect size estimate for each SNP
- `se`: 982 by 1 vector, standard errors of the single-SNP effect size estimates
- `Nsnp`: 982 by 1 vector, sample size of each SNP
- `cohort_R`: cohort sample LD
- `shrink_R`: shrinkage panel sample LD
- `panel_R`: panel sample LD
- `snp_info`: 3 by 1 cell, ID and allele of each SNP

**Step 2**. Fit three RSS-BVSR models with different LD matrices.

```matlab
% cohort sample LD
[betasam, gammasam, hsam, logpisam, Naccept] = rss_bvsr(betahat, se, cohort_R, Nsnp, Ndraw, Nburn, Nthin);

% shrinkage panel sample LD
[betasam, gammasam, hsam, logpisam, Naccept] = rss_bvsr(betahat, se, shrink_R, Nsnp, Ndraw, Nburn, Nthin);

% panel sample LD
[betasam, gammasam, hsam, logpisam, Naccept] = rss_bvsr(betahat, se, panel_R, Nsnp, Ndraw, Nburn, Nthin);
```

## More simulations

The simulations in Section 4.1 of [Zhu and Stephens (2017)][]
are essentially "replications" of the example above.
The simulated datasets in Section 4.1 are available as `rss_example2_data_{1/2/3}.tar.gz`<sup>1</sup>.

Each simulated dataset contains three files:
`genotype.txt`, `phenotype.txt` and `simulated_data.mat`.
The files `genotype.txt` and `phenotype.txt` are the genotype and phenotype files for
[`GEMMA`](https://github.com/xiangzhou/GEMMA).
The file `simulated_data.mat` contains three cells.

```matlab
true_para = {pve, beta, gamma, sigma};
individual_data = {y, X};
summary_data = {betahat, se, Nsnp};
```

Only the `summary_data` cell above is used as the input for RSS methods.

RSS methods also require an estimated LD matrix.
The three types of LD matrices are provided in the file `genotype2.mat`<sup>1</sup>.

After applying RSS methods to these simulated data,
we obtain the following results.

| True PVE = 0.2 |
|:-:|
|<img src="images/LD1.png" width="600"> |

| True PVE = 0.02 | True PVE = 0.002 | 
|:-:|:-:|
|<img src="images/LD2.png" width="400"> | <img src="images/LD3.png" width="400"> | 

--------

**Footnotes:**

1. Currently these files are locked, since they contain individual-level genotypes
from Wellcome Trust Case Control Consortium (WTCCC, <https://www.wtccc.org.uk/>).
You need to get permission from WTCCC before we can share these files with you.
