---
title: "Tutorial Catalog"
author: Xiang Zhu
output:
  workflowr::wflow_html:
    toc: false
editor_options:
  chunk_output_type: console
---

[Zhu and Stephens (2017)]: https://dx.doi.org/10.1214/17-AOAS1046
[Zhu and Stephens (2018)]: https://doi.org/10.1038/s41467-018-06805-x
[Zhu et al (2021)]: https://doi.org/10.1038/s41467-021-22588-0

We illustrate how to use RSS methods and software through the following examples.

### Example 1: Fit RSS models via MCMC, and estimate SNP heritability (PVE).

[Example 1](example_1.html) illustrates how to fit RSS models using MCMC
algorithms. Three types of prior distributions are considered:
BVSR [(Guan and Stephens, 2011)](https://projecteuclid.org/euclid.aoas/1318514285),
BSLMM [(Zhou et al, 2013)](https://doi.org/10.1371/journal.pgen.1003264),
and ASH [(Stephens, 2017)](https://doi.org/10.1093/biostatistics/kxw041).
The MCMC output is further used to estimate the SNP heritability.
This example is closely related to Section 4.2 of [Zhu and Stephens (2017)][]. 

### Example 2: Fit RSS-BVSR via MCMC with three types of LD matrix.

[Example 2](example_2.html) illustrates the impact of different LD estimates on the
RSS results. Three types of estimated LD matrices are considered:
sample LD based on cohort individuals, sample LD based on panel individuals,
and shrinkage LD estimate based on panel individuals
[(Wen and Stephens, 2010)](https://www.ncbi.nlm.nih.gov/pubmed/21479081).
This example is closely related to Section 4.1 of [Zhu and Stephens (2017)][]. 

### Example 3: Fit RSS-BVSR via MCMC with two types of standard error (SE).

[Example 3](example_3.html) illustrates the impact of two definitions of SE on RSS results.
This example is closely related to Section 2.1 of [Zhu and Stephens (2017)][].  

### Example 4: Fit RSS-BVSR via variational Bayes (VB) methods.

[Example 4](example_4.html) illustrates how to fit an RSS-BVSR model using
variational Bayes (VB) approximation, and compares the results with
previous work based on individual-level data
[(Carbonetto and Stephens, 2012)](https://projecteuclid.org/euclid.ba/1339616726).
This example is closely related to the section entitled
"Connection with enrichment analysis of individual-level data" of [Zhu and Stephens (2018)][].

### Example 5: Enrichment and prioritization analysis based on RSS-E.

[Example 5](example_5.html) illustrates how to perform
enrichment and prioritization analysis of GWAS summary statistics
based on RSS-E. This example consists of two parts:

#### [Example 5, Part A](example_5a.html)

This part shows enrichment and prioritization analysis of a synthetic dataset
used in simulation studies of [Zhu and Stephens (2018)][].
Part A gives users a quick view of how RSS-E works in enrichment and prioritization analysis.

#### [Example 5, Part B](example_5b.html)

This part shows an end-to-end enrichment and prioritization analysis of
inflammatory bowel disease GWAS summary statistics
[(Liu et al, 2015)](https://www.ncbi.nlm.nih.gov/pubmed/26192919)
and a gene set named "IL23-mediated signaling events" (Pathway Commons 2, PID, 37 genes) based on RSS-E.
Part B illustrates the actual data analyses performed in [Zhu and Stephens (2018)][].

### Example 6: Enrichment and prioritization analysis based on RSS-NET.

[Example 6](https://suwonglab.github.io/rss-net/) illustrates
how to perform enrichment and prioritization analysis of GWAS
summary statistics based on RSS-NET. This example consists of two parts:

#### [Example 6, Part A](https://suwonglab.github.io/rss-net/wtccc_bcell.html)

This part shows enrichment and prioritization analysis of two synthetic datasets
used in simulation studies of [Zhu et al (2021)][].
Part A gives users a quick view of how RSS-NET works in enrichment and prioritization analysis.

#### [Example 6, Part B](https://suwonglab.github.io/rss-net/ibd2015_nkcell.html)

This part shows an end-to-end enrichment and prioritization analysis of
inflammatory bowel disease GWAS summary statistics
[(Liu et al, 2015)](https://www.ncbi.nlm.nih.gov/pubmed/26192919)
and a gene regulatory network for natural killer cells
[(ENCODE Project Consortium, 2012)](https://pubmed.ncbi.nlm.nih.gov/22955616) based on RSS-NET.
Part B illustrates the actual data analyses performed in [Zhu et al (2021)][].
