<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Xiang Zhu" />


<title>Example 1: RSS Model Fitting via MCMC &amp; SNP Heritability Estimation</title>

<script src="site_libs/header-attrs-2.28/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">rss</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="setup.html">Setup</a>
</li>
<li>
  <a href="function.html">Functions</a>
</li>
<li>
  <a href="example.html">Examples</a>
</li>
<li>
  <a href="faq.html">FAQs</a>
</li>
<li>
  <a href="news.html">News</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephenslab/rss">
    <span class="fab fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Example 1: RSS Model Fitting via MCMC &amp;
SNP Heritability Estimation</h1>
<h4 class="author">Xiang Zhu</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-09-16
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>rss/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200623code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20200623)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200623code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200623)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabrsstree941a146ebc0cf0ff1dfa5bbd4f2662140b4dddaetargetblank941a146a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/stephenslab/rss/tree/941a146ebc0cf0ff1dfa5bbd4f2662140b4dddae" target="_blank">941a146</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabrsstree941a146ebc0cf0ff1dfa5bbd4f2662140b4dddaetargetblank941a146a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/stephenslab/rss/tree/941a146ebc0cf0ff1dfa5bbd4f2662140b4dddae" target="_blank">941a146</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>rmd/example_1.Rmd</code>) and HTML
(<code>docs/example_1.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/rss/985fc1cb3c38e9696eb9a92a293d6d0b8ad9cd7a/docs/example_1.html" target="_blank">985fc1c</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2024-07-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/rss/blob/5602ceb4d0b1922d94bb4dae74a2aecb57d9c15e/rmd/example_1.Rmd" target="_blank">5602ceb</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2024-07-03
</td>
<td>
wflow_publish("rmd/example_1.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/rss/6e9c8f08e82996ee1364f65dec491d84b6bcac26/docs/example_1.html" target="_blank">6e9c8f0</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2021-03-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/rss/blob/17874858b0537e7da55972e7cc6cc5bb6330de21/rmd/example_1.Rmd" target="_blank">1787485</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2021-03-05
</td>
<td>
wflow_publish("rmd/example_1.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/rss/bab3f582716fec43ec1d9285b4aff4484afb6075/docs/example_1.html" target="_blank">bab3f58</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2020-06-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/rss/348801405cbc75da2da82682089a960359a43103/docs/example_1.html" target="_blank">3488014</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2020-06-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/rss/blob/221b7e369b3234032281cbac53cff44fae285410/rmd/example_1.Rmd" target="_blank">221b7e3</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2020-06-23
</td>
<td>
wflow_publish("rmd/example_1.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="overview" class="section level2">
<h2>Overview</h2>
<p>This example illustrates how to fit RSS models using MCMC algorithms.
Three types of prior distributions are considered: BVSR in <a
href="https://projecteuclid.org/euclid.aoas/1318514285">Guan and
Stephens (2011)</a>, BSLMM in <a
href="https://doi.org/10.1371/journal.pgen.1003264">Zhou, Carbonetto and
Stephens (2013)</a>, and ASH in <a
href="https://doi.org/10.1093/biostatistics/kxw041">Stephens (2017)</a>.
The MCMC output is further used to estimate the SNP heritability. This
example is closely related to Section 4.2 of <a
href="https://projecteuclid.org/euclid.aoas/1507168840">Zhu and Stephens
(2017)</a>.</p>
<p>The single-SNP summary-level data are computed from a simulated GWAS
dataset. The GWAS data are simulated under the Scenario 2.1 in <a
href="https://projecteuclid.org/euclid.aoas/1507168840">Zhu and Stephens
(2017)</a>. Specifically, 100 “causal” SNPs are randomly drawn from
12758 SNPs on chromosome 16, with effect sizes coming from standard
normal <span class="math inline">\({\cal N}(0,1)\)</span>. Effect sizes
of remaining SNPs are zero. The true PVE (SNP heritability) is 0.2.</p>
<p>To reproduce results of Example 1, please read the step-by-step guide
below and run <a
href="https://github.com/stephenslab/rss/blob/master/examples/example1.m"><code>example1.m</code></a>.
Before running <a
href="https://github.com/stephenslab/rss/blob/master/examples/example1.m"><code>example1.m</code></a>,
please first install the <a
href="https://github.com/stephenslab/rss/tree/master/src">MCMC
subroutines</a>. Please find installation instructions <a
href="setup_mcmc.html">here</a>.</p>
</div>
<div id="step-by-step-illustration" class="section level2">
<h2>Step-by-step illustration</h2>
<p><strong>Step 1</strong>. Download data files.</p>
<p>All data files required to run this example are freely available at
Zenodo <a href="https://doi.org/10.5281/zenodo.12636265"><img
src="https://zenodo.org/badge/DOI/10.5281/zenodo.12636265.svg"
alt="DOI" /></a>. Please contact me if you have trouble accessing this
file. After a complete download, you should see the following files.</p>
<div class="sourceCode" id="cb1"><pre
class="sourceCode zsh"><code class="sourceCode zsh"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="ex">$</span> tree ./</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="ex">./</span></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="ex">├──</span> example1.mat</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="ex">├──</span> example1_rssash.mat</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="ex">├──</span> example1_rssbslmm.mat</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="ex">├──</span> example1_rssbvsr.mat</span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="ex">└──</span> readme</span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a></span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a><span class="ex">0</span> directories, 5 files</span></code></pre></div>
<p>The data file <code>example1.mat</code> contains the following
elements.</p>
<ul>
<li><code>betahat</code>: 12758 by 1 vector, single-SNP effect size
estimate for each SNP</li>
<li><code>se</code>: 12758 by 1 vector, standard errors of the
single-SNP effect size estimates</li>
<li><code>Nsnp</code>: 12758 by 1 vector, sample size of each SNP</li>
<li><code>R</code>: 12758 by 12758 matrix, LD matrix estimated from a
reference panel</li>
<li><code>bwd</code>: integer, bandwidth of the banded matrix
<code>R</code></li>
<li><code>BR</code>: (<code>bwd</code>+1) by 12758 matrix, banded
storage of matrix <code>R</code></li>
</ul>
<p>Note that only <code>betahat</code>, <code>se</code>,
<code>Nsnp</code> and <code>R</code> are needed for running MCMC. The
other two quantities, <code>bwd</code> and <code>BR</code>, are only
used in SNP heritability calculation.</p>
<p><strong>Step 2</strong>. Check the “small effects” model
assumption.</p>
<p>Using single-SNP summary data, we can compute squared sample
correlation between phenotype and each SNP, and then check the “small
effect” assumption by looking at these marginal squared correlation
values (please see Table 1 of <a
href="https://projecteuclid.org/euclid.aoas/1507168840">Zhu and Stephens
(2017)</a> for more details). The following code illustrates the “small
effect” check in <a
href="https://github.com/stephenslab/rss/blob/master/examples/example1.m"><code>example1.m</code></a><sup>1</sup>.</p>
<div class="sourceCode" id="cb2"><pre
class="sourceCode matlab"><code class="sourceCode matlab"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="op">&gt;&gt;</span> <span class="va">chatsqr</span> <span class="op">=</span> (<span class="va">betahat</span>(<span class="op">:</span>)<span class="op">.^</span><span class="fl">2</span>) <span class="op">./</span> (<span class="va">Nsnp</span>(<span class="op">:</span>)<span class="op">.*</span>(<span class="va">se</span>(<span class="op">:</span>)<span class="op">.^</span><span class="fl">2</span>) <span class="op">+</span> <span class="va">betahat</span>(<span class="op">:</span>)<span class="op">.^</span><span class="fl">2</span>)<span class="op">;</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="op">&gt;&gt;</span> <span class="va">disp</span>(<span class="va">prctile</span>(<span class="va">log10</span>(<span class="va">chatsqr</span>)<span class="op">,</span> <span class="fl">0</span><span class="op">:</span><span class="fl">25</span><span class="op">:</span><span class="fl">100</span>))<span class="op">;</span></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>  <span class="op">-</span><span class="fl">11.6029</span>   <span class="op">-</span><span class="fl">4.1154</span>   <span class="op">-</span><span class="fl">3.4721</span>   <span class="op">-</span><span class="fl">2.9962</span>   <span class="op">-</span><span class="fl">1.5982</span></span></code></pre></div>
<p>Since our data are generated from genotypes of a single chromosome,
the simulated effect sizes per SNP are larger than would be expected in
a typical GWAS (see Table 1 <a
href="https://projecteuclid.org/euclid.aoas/1507168840">Zhu and Stephens
(2017)</a>).</p>
<p><strong>Step 3</strong>. Fit RSS-BVSR, RSS-BSLMM and RSS-ASH models
via MCMC.</p>
<p>To fit RSS-BVSR and RSS-BSLMM models, we only need to specify the
length of Markov chains for RSS software.</p>
<div class="sourceCode" id="cb3"><pre
class="sourceCode matlab"><code class="sourceCode matlab"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="va">Ndraw</span> <span class="op">=</span> <span class="fl">2e6</span><span class="op">;</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="va">Nburn</span> <span class="op">=</span> <span class="fl">2e5</span><span class="op">;</span></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="va">Nthin</span> <span class="op">=</span> <span class="fl">9e1</span><span class="op">;</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>[<span class="va">betasam</span><span class="op">,</span> <span class="va">gammasam</span><span class="op">,</span> <span class="va">hsam</span><span class="op">,</span> <span class="va">logpisam</span><span class="op">,</span> <span class="va">Naccept</span>] <span class="op">=</span> <span class="va">rss_bvsr</span>(<span class="va">betahat</span><span class="op">,</span> <span class="va">se</span><span class="op">,</span> <span class="va">R</span><span class="op">,</span> <span class="va">Nsnp</span><span class="op">,</span> <span class="va">Ndraw</span><span class="op">,</span> <span class="va">Nburn</span><span class="op">,</span> <span class="va">Nthin</span>)<span class="op">;</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>[<span class="va">bsam</span><span class="op">,</span> <span class="va">zsam</span><span class="op">,</span> <span class="va">lpsam</span><span class="op">,</span> <span class="va">hsam</span><span class="op">,</span> <span class="va">rsam</span><span class="op">,</span> <span class="va">Naccept</span>] <span class="op">=</span> <span class="va">rss_bslmm</span>(<span class="va">betahat</span><span class="op">,</span> <span class="va">se</span><span class="op">,</span> <span class="va">R</span><span class="op">,</span> <span class="va">Nsnp</span><span class="op">,</span> <span class="va">Ndraw</span><span class="op">,</span> <span class="va">Nburn</span><span class="op">,</span> <span class="va">Nthin</span>)<span class="op">;</span></span></code></pre></div>
<p>To fit RSS-ASH model, we need to specify the length of Markov chain
and a grid for the prior standard deviations of multiple-SNP effect
sizes.</p>
<div class="sourceCode" id="cb4"><pre
class="sourceCode matlab"><code class="sourceCode matlab"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="va">Ndraw</span> <span class="op">=</span> <span class="fl">5e7</span><span class="op">;</span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="va">Nburn</span> <span class="op">=</span> <span class="fl">1e7</span><span class="op">;</span></span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="va">Nthin</span> <span class="op">=</span> <span class="fl">1e3</span><span class="op">;</span></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="va">sigma_beta</span> <span class="op">=</span> [<span class="fl">0</span> <span class="fl">0.001</span> <span class="fl">0.003</span> <span class="fl">0.01</span> <span class="fl">0.03</span> <span class="fl">0.1</span> <span class="fl">0.3</span> <span class="fl">1</span> <span class="fl">3</span>]<span class="op">;</span></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>[<span class="va">bsam</span><span class="op">,</span> <span class="va">zsam</span><span class="op">,</span> <span class="va">wsam</span><span class="op">,</span> <span class="va">lsam</span><span class="op">,</span> <span class="va">Naccept</span>] <span class="op">=</span> <span class="va">rss_ash</span>(<span class="va">betahat</span><span class="op">,</span> <span class="va">se</span><span class="op">,</span> <span class="va">R</span><span class="op">,</span> <span class="va">Nsnp</span><span class="op">,</span> <span class="va">sigma_beta</span><span class="op">,</span> <span class="va">Ndraw</span><span class="op">,</span> <span class="va">Nburn</span><span class="op">,</span> <span class="va">Nthin</span>)<span class="op">;</span></span></code></pre></div>
<p><strong>Step 4</strong>. Estimate SNP heritability.</p>
<p>We use the posterior sample of multiple-SNP effect sizes
(<code>bsam</code>) to obtain the posterior sample of SNP heritability
(<code>pvesam</code>).</p>
<div class="sourceCode" id="cb5"><pre
class="sourceCode matlab"><code class="sourceCode matlab"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="va">M</span> <span class="op">=</span> <span class="va">length</span>(<span class="va">hsam</span>)<span class="op">;</span> <span class="co">% the length of posterior simulations</span></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="va">pvesam</span> <span class="op">=</span> <span class="va">zeros</span>(<span class="va">M</span><span class="op">,</span><span class="fl">1</span>)<span class="op">;</span> <span class="co">% preallocate the pve posterior estimates</span></span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="kw">for</span> <span class="va">i</span> <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="va">M</span> </span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>  <span class="va">pvesam</span>(<span class="va">i</span>) <span class="op">=</span> <span class="va">compute_pve</span>(<span class="va">bsam</span>(<span class="va">i</span><span class="op">,:</span>)<span class="op">,</span> <span class="va">betahat</span><span class="op">,</span> <span class="va">se</span><span class="op">,</span> <span class="va">Nsnp</span><span class="op">,</span> <span class="va">bwd</span><span class="op">,</span> <span class="va">BR</span><span class="op">,</span> <span class="fl">1</span>)<span class="op">;</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a><span class="kw">end</span></span></code></pre></div>
<p>Recall that the SNP heritability estimator (Equation 3.8 of <a
href="https://projecteuclid.org/euclid.aoas/1507168840">Zhu and Stephens
(2017)</a>) involves vector-matrix-vector product. To speed calculation,
we exploit the banded structure of <code>R</code> and use the banded
version of vector-matrix-vector product implemented in
<code>lapack</code>. Hence, the banded storage <code>BR</code>, instead
of the original form <code>R</code>, is used to calculate SNP
heritability.</p>
<p><strong>Step 5</strong>. Summarize results.</p>
<p>The dataset is simulated with the true SNP heritability (PVE) being
0.2. The following table summarizes the posterior estimates (with 95%
credible interval) and the total computational time (including MCMC
iterations and PVE calculations) for three models.</p>
<table>
<thead>
<tr class="header">
<th align="center">Model</th>
<th align="center">PVE estimation</th>
<th align="center">Total time</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">RSS-BVSR</td>
<td align="center">0.200 [0.125, 0.290]</td>
<td align="center">1.38 hours</td>
</tr>
<tr class="even">
<td align="center">RSS-BSLMM</td>
<td align="center">0.216 [0.136, 0.306]</td>
<td align="center">2.52 hours</td>
</tr>
<tr class="odd">
<td align="center">RSS-ASH</td>
<td align="center">0.197 [0.114, 0.286]</td>
<td align="center">6.69 hours</td>
</tr>
</tbody>
</table>
<p>The following histograms show the posterior distributions of
estimated SNP heritability under these three models.</p>
<center>
<div class="float">
<img src="images/rss_example1_pve.png" alt="example1_pve" />
<div class="figcaption">example1_pve</div>
</div>
</center>
</div>
<div id="more-simulations" class="section level2">
<h2>More simulations</h2>
<p>Simulations in Section 4.2 of <a
href="https://projecteuclid.org/euclid.aoas/1507168840">Zhu and Stephens
(2017)</a> are essentially “replications” of the example above. The
simulated datasets for Scenarios 2.1 and 2.2 in Section 4.2 of <a
href="https://projecteuclid.org/euclid.aoas/1507168840">Zhu and Stephens
(2017)</a> are available as
<code>rss_example1_simulations.tar.gz</code><sup>2</sup>.</p>
<p>Each simulated dataset contains three files:
<code>genotype.txt</code>, <code>phenotype.txt</code> and
<code>simulated_data.mat</code>. The files <code>genotype.txt</code> and
<code>phenotype.txt</code> are the genotype and phenotype files for <a
href="https://github.com/xiangzhou/GEMMA"><code>GEMMA</code></a>. The
file <code>simulated_data.mat</code> contains three cells.</p>
<div class="sourceCode" id="cb6"><pre
class="sourceCode matlab"><code class="sourceCode matlab"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="va">true_para</span> <span class="op">=</span> {<span class="va">pve</span><span class="op">,</span> <span class="va">beta</span><span class="op">,</span> <span class="va">gamma</span><span class="op">,</span> <span class="va">sigma</span>}<span class="op">;</span></span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="va">individual_data</span> <span class="op">=</span> {<span class="va">y</span><span class="op">,</span> <span class="va">X</span>}<span class="op">;</span></span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a><span class="va">summary_data</span> <span class="op">=</span> {<span class="va">betahat</span><span class="op">,</span> <span class="va">se</span><span class="op">,</span> <span class="va">Nsnp</span>}<span class="op">;</span></span></code></pre></div>
<p>Only the <code>summary_data</code> cell above is used as the input
for RSS methods.</p>
<p>The RSS methods also require an estimated LD matrix <code>R</code>.
This matrix <code>R</code> is provided in the file
<code>genotype.mat</code>.</p>
<p>After applying RSS methods to these simulated data, we obtain the
following PVE estimation results.</p>
<table>
<colgroup>
<col width="50%" />
<col width="50%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Scenario 2.1 (sparse), True PVE = 0.2</th>
<th align="center">Scenario 2.1 (sparse), True PVE = 0.6</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center"><img src="images/pve2sparse_1.png" width="400"></td>
<td align="center"><img src="images/pve2sparse_2.png" width="400"></td>
</tr>
</tbody>
</table>
<table>
<colgroup>
<col width="50%" />
<col width="50%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Scenario 2.2 (polygenic), True PVE = 0.2</th>
<th align="center">Scenario 2.2 (polygenic), True PVE = 0.6</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td
align="center"><img src="images/pve2polygenic_1.png" width="400"></td>
<td
align="center"><img src="images/pve2polygenic_2.png" width="400"></td>
</tr>
</tbody>
</table>
<hr />
<p><strong>Footnotes:</strong></p>
<ol style="list-style-type: decimal">
<li><p>The function <a
href="http://www.mathworks.com/help/stats/prctile.html"><code>prctile</code></a>
used in <a
href="https://github.com/stephenslab/rss/blob/master/examples/example1.m"><code>example1.m</code></a>
requires the <a
href="http://www.mathworks.com/help/stats/index.html">Statistics and
Machine Learning Toolbox</a>. Please see this <a
href="https://github.com/stephenslab/rss/pull/3/commits/566e149ed840a913bfef9c0d7bf82feb41d6735d">commit</a>
(courtesy of <a href="https://pcarbo.github.io/">Dr. Peter
Carbonetto</a>) if this required toolbox is not available in your
environment.</p></li>
<li><p>Currently these files are locked, since they contain
individual-level genotypes from Wellcome Trust Case Control Consortium
(WTCCC, <a href="https://www.wtccc.org.uk/"
class="uri">https://www.wtccc.org.uk/</a>). You need to get permission
from WTCCC before we can share these files with you.</p></li>
</ol>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">session_info</span>()</span></code></pre></div>
<pre><code>─ Session info ───────────────────────────────────────────────────────────────
 setting  value
 version  R version 4.4.1 (2024-06-14)
 os       macOS Sonoma 14.6.1
 system   aarch64, darwin20
 ui       X11
 language (EN)
 collate  en_US.UTF-8
 ctype    en_US.UTF-8
 tz       America/Los_Angeles
 date     2024-09-16
 pandoc   3.1.11 @ /Applications/RStudio.app/Contents/Resources/app/quarto/bin/tools/aarch64/ (via rmarkdown)

─ Packages ───────────────────────────────────────────────────────────────────
 package     * version    date (UTC) lib source
 bslib         0.8.0      2024-07-29 [1] CRAN (R 4.4.0)
 cachem        1.1.0      2024-05-16 [1] CRAN (R 4.4.0)
 callr         3.7.6      2024-03-25 [1] CRAN (R 4.4.0)
 cli           3.6.3      2024-06-21 [1] CRAN (R 4.4.0)
 devtools      2.4.5      2022-10-11 [1] CRAN (R 4.4.0)
 digest        0.6.37     2024-08-19 [1] CRAN (R 4.4.1)
 ellipsis      0.3.2      2021-04-29 [1] CRAN (R 4.4.0)
 evaluate      0.24.0     2024-06-10 [1] CRAN (R 4.4.0)
 fansi         1.0.6      2023-12-08 [1] CRAN (R 4.4.0)
 fastmap       1.2.0      2024-05-15 [1] CRAN (R 4.4.0)
 fs            1.6.4      2024-04-25 [1] CRAN (R 4.4.0)
 getPass       0.2-4      2023-12-10 [1] CRAN (R 4.4.0)
 git2r         0.33.0     2023-11-26 [1] CRAN (R 4.4.0)
 glue          1.7.0      2024-01-09 [1] CRAN (R 4.4.0)
 htmltools     0.5.8.1    2024-04-04 [1] CRAN (R 4.4.0)
 htmlwidgets   1.6.4      2023-12-06 [1] CRAN (R 4.4.0)
 httpuv        1.6.15     2024-03-26 [1] CRAN (R 4.4.0)
 httr          1.4.7      2023-08-15 [1] CRAN (R 4.4.0)
 jquerylib     0.1.4      2021-04-26 [1] CRAN (R 4.4.0)
 jsonlite      1.8.8      2023-12-04 [1] CRAN (R 4.4.0)
 knitr         1.48       2024-07-07 [1] CRAN (R 4.4.0)
 later         1.3.2      2023-12-06 [1] CRAN (R 4.4.0)
 lifecycle     1.0.4      2023-11-07 [1] CRAN (R 4.4.0)
 magrittr      2.0.3      2022-03-30 [1] CRAN (R 4.4.0)
 memoise       2.0.1      2021-11-26 [1] CRAN (R 4.4.0)
 mime          0.12       2021-09-28 [1] CRAN (R 4.4.0)
 miniUI        0.1.1.1    2018-05-18 [1] CRAN (R 4.4.0)
 pillar        1.9.0      2023-03-22 [1] CRAN (R 4.4.0)
 pkgbuild      1.4.4      2024-03-17 [1] CRAN (R 4.4.0)
 pkgconfig     2.0.3      2019-09-22 [1] CRAN (R 4.4.0)
 pkgload       1.4.0      2024-06-28 [1] CRAN (R 4.4.0)
 processx      3.8.4      2024-03-16 [1] CRAN (R 4.4.0)
 profvis       0.3.8      2023-05-02 [1] CRAN (R 4.4.0)
 promises      1.3.0      2024-04-05 [1] CRAN (R 4.4.0)
 ps            1.8.0      2024-09-12 [1] CRAN (R 4.4.1)
 purrr         1.0.2      2023-08-10 [1] CRAN (R 4.4.0)
 R6            2.5.1      2021-08-19 [1] CRAN (R 4.4.0)
 Rcpp          1.0.13     2024-07-17 [1] CRAN (R 4.4.0)
 remotes       2.5.0      2024-03-17 [1] CRAN (R 4.4.0)
 rlang         1.1.4      2024-06-04 [1] CRAN (R 4.4.0)
 rmarkdown     2.28       2024-08-17 [1] CRAN (R 4.4.0)
 rprojroot     2.0.4      2023-11-05 [1] CRAN (R 4.4.0)
 rstudioapi    0.16.0     2024-03-24 [1] CRAN (R 4.4.0)
 sass          0.4.9.9000 2024-07-11 [1] Github (rstudio/sass@9228fcf)
 sessioninfo   1.2.2      2021-12-06 [1] CRAN (R 4.4.0)
 shiny         1.9.1      2024-08-01 [1] CRAN (R 4.4.0)
 stringi       1.8.4      2024-05-06 [1] CRAN (R 4.4.0)
 stringr       1.5.1      2023-11-14 [1] CRAN (R 4.4.0)
 tibble        3.2.1      2023-03-20 [1] CRAN (R 4.4.0)
 urlchecker    1.0.1      2021-11-30 [1] CRAN (R 4.4.0)
 usethis       3.0.0      2024-07-29 [1] CRAN (R 4.4.0)
 utf8          1.2.4      2023-10-22 [1] CRAN (R 4.4.0)
 vctrs         0.6.5      2023-12-01 [1] CRAN (R 4.4.0)
 whisker       0.4.1      2022-12-05 [1] CRAN (R 4.4.0)
 workflowr   * 1.7.1      2023-08-23 [1] CRAN (R 4.4.0)
 xfun          0.47       2024-08-17 [1] CRAN (R 4.4.0)
 xtable        1.8-4      2019-04-21 [1] CRAN (R 4.4.0)
 yaml          2.3.10     2024-07-26 [1] CRAN (R 4.4.0)

 [1] /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/library

──────────────────────────────────────────────────────────────────────────────</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
