<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Xiang Zhu" />


<title>Frequently Asked Questions</title>

<script src="site_libs/header-attrs-2.28/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>










<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">rss</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="setup.html">Setup</a>
</li>
<li>
  <a href="function.html">Functions</a>
</li>
<li>
  <a href="example.html">Examples</a>
</li>
<li>
  <a href="faq.html">FAQs</a>
</li>
<li>
  <a href="news.html">News</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephenslab/rss">
    <span class="fab fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Frequently Asked Questions</h1>
<h4 class="author">Xiang Zhu</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-09-16
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>rss/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabrsstree941a146ebc0cf0ff1dfa5bbd4f2662140b4dddaetargetblank941a146a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/stephenslab/rss/tree/941a146ebc0cf0ff1dfa5bbd4f2662140b4dddae" target="_blank">941a146</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabrsstree941a146ebc0cf0ff1dfa5bbd4f2662140b4dddaetargetblank941a146a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/stephenslab/rss/tree/941a146ebc0cf0ff1dfa5bbd4f2662140b4dddae" target="_blank">941a146</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>rmd/faq.Rmd</code>) and HTML
(<code>docs/faq.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/rss/ef0166b9a6d7eee80107d5fca069e9b51673f0a6/docs/faq.html" target="_blank">ef0166b</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2022-01-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/rss/blob/0e29090d92db7a4fd73af9e7e74ac230d3bcec06/rmd/faq.Rmd" target="_blank">0e29090</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2022-01-24
</td>
<td>
wflow_publish("rmd/faq.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/rss/f405f7d998b4fe2ce742c0a04fe6517c901fedaa/docs/faq.html" target="_blank">f405f7d</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2021-03-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/rss/blob/7a309897081ec531f842ea8ea0a5c3ffa354179c/rmd/faq.Rmd" target="_blank">7a30989</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2021-03-05
</td>
<td>
wflow_publish("rmd/faq.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/rss/e66733fd6b158797aa13a844972e3f988420eae3/docs/faq.html" target="_blank">e66733f</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2020-06-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/rss/blob/3a6660a45719d8d980e19e12bbaae8d79564a000/rmd/faq.Rmd" target="_blank">3a6660a</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2020-06-24
</td>
<td>
wflow_publish("rmd/faq.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/rss/bab3f582716fec43ec1d9285b4aff4484afb6075/docs/faq.html" target="_blank">bab3f58</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2020-06-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/rss/68f2b9b06a5ebb0e9ac7315773867db0be0bb664/docs/faq.html" target="_blank">68f2b9b</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2020-06-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/rss/blob/d40657cb2ab2c4bf88c7435e18c03da69afcbbde/rmd/faq.Rmd" target="_blank">d40657c</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2020-06-23
</td>
<td>
wflow_publish("rmd/faq.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Please feel free to create a new <a
href="https://github.com/stephenslab/rss/issues">issue</a> and/or send
an email to <a
href="https://github.com/xiangzhu"><code>xiangzhu[at]psu.edu</code></a>.</p>
<div id="statistics" class="section level2">
<h2>Statistics</h2>
<blockquote>
<p>Q: The following distribution is frequently used in RSS methods:
<span class="math inline">\(\beta_j \sim \pi_j\cdot{\cal
N}(0,\sigma_j^2) + (1-\pi_j)\cdot\delta_0\)</span>. How do you simulate
samples from this distribution?</p>
</blockquote>
<p>A: This is a mixture of a normal distribution <span
class="math inline">\({\cal N}(0,\sigma_j^2)\)</span> and a point mass
at zero <span class="math inline">\(\delta_0\)</span>. To simulate
numbers from this mixture distribution, we can first simulate a latent
variable <span class="math inline">\(\gamma_j\)</span> from a Bernoulli
distribution which takes the value 1 with probability <span
class="math inline">\(\pi_j\)</span>. If <span
class="math inline">\(\gamma_j=1\)</span>, we draw the value of <span
class="math inline">\(\beta_j\)</span> from the normal distribution
<span class="math inline">\({\cal N}(0,\sigma_j^2)\)</span>. If <span
class="math inline">\(\gamma_j=0\)</span>, we set <span
class="math inline">\(\beta_j=0\)</span>.</p>
<blockquote>
<p>Q: The RSS-BVSR model contains two parameters <span
class="math inline">\(\{\pi,\sigma_B^2\}\)</span>: <span
class="math inline">\(\beta_j \sim \pi\cdot{\cal N}(0,\sigma_B^2) +
(1-\pi)\cdot\delta_0\)</span>. How do you use the output of <a
href="https://github.com/stephenslab/rss/blob/master/src/rss_bvsr.m"><code>rss_bvsr.m</code></a>
to find the posterior means of <span
class="math inline">\(\{\pi,\sigma_B^2\}\)</span>?</p>
</blockquote>
<p>A: To find posterior means of <span
class="math inline">\(\{\pi,\sigma_B^2\}\)</span>, we need to obtain
their MCMC samples from the output of <a
href="https://github.com/stephenslab/rss/blob/master/src/rss_bvsr.m"><code>rss_bvsr.m</code></a>
first. The MCMC samples of <span class="math inline">\(\pi\)</span> is
simply <code>exp(logpisam)</code>, where <code>logpisam</code> is the
MCMC samples of <span class="math inline">\(\log(\pi)\)</span> from the
output of <a
href="https://github.com/stephenslab/rss/blob/master/src/rss_bvsr.m"><code>rss_bvsr.m</code></a>.
The MCMC samples of <span class="math inline">\(\sigma_B^2\)</span> is
obtained as follows: (1) Get the MCMC samples of <span
class="math inline">\(\{h,\pi\}\)</span> from the output of <a
href="https://github.com/stephenslab/rss/blob/master/src/rss_bvsr.m"><code>rss_bvsr.m</code></a>
as <code>hsam</code> and <code>exp(logpisam)</code>; (2) Use
<code>hsam</code> and <code>exp(logpisam)</code> to obtain the MCMC
samples of <span class="math inline">\(\sigma_B^2\)</span>, based on the
first half of Equation 3.4 in <a
href="https://www.ncbi.nlm.nih.gov/pubmed/29399241">Zhu and Stephens
(2017)</a> (NB: <span class="math inline">\(\rho\equiv 1\)</span> for
RSS-BVSR). Once we have the MCMC samples of <span
class="math inline">\(\{\pi,\sigma_B^2\}\)</span>, we can compute their
sample means to estimate posterior means of <span
class="math inline">\(\{\pi,\sigma_B^2\}\)</span>.</p>
<blockquote>
<p>Q: How could I reduce the computational time of Markov chain Monte
Carlo (MCMC) programs provided in <a
href="https://www.ncbi.nlm.nih.gov/pubmed/29399241">Zhu and Stephens
(2017)</a>?</p>
</blockquote>
<p>A: You can reduce the time by shortening the length of MCMC chains,
that is, setting a smaller value for <code>Ndraw</code> in the MCMC
programs (<a
href="https://github.com/stephenslab/rss/tree/master/src"><code>rss/src</code></a>).
However, please note that using short MCMC chains may yield inaccurate
inference results for large-scale problems.</p>
<blockquote>
<p>Q: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29399241">Zhu and
Stephens (2017)</a> reported a leave-one-out residual imputation method
for model checking. Do you have software for this method?</p>
</blockquote>
<p>A: Yes. I write a subroutine <a
href="https://github.com/stephenslab/rss/blob/master/misc/mvnloo.m"><code>mvnloo.m</code></a>
for this check. This leave-one-out residual imputation is a simple
application of multivariate normal distribution. For more details,
please see Equations D.1 and D.2 on page 10 of <a
href="http://stephenslab.uchicago.edu/assets/papers/Zhu2017-supplement.pdf">Supplement
to Zhu and Stephens (2017)</a>.</p>
<blockquote>
<p>Q: How could I reduce the computational time of variational Bayes
(VB) programs provided in <a
href="https://www.ncbi.nlm.nih.gov/pubmed/30341297">Zhu and Stephens
(2018)</a>?</p>
</blockquote>
<p>A: There are two options to reduce the time of running the VB
programs (<a
href="https://github.com/stephenslab/rss/tree/master/src_vb"><code>rss/src_vb</code></a>).
First, you can set a time limit for the VB programs, that is, setting
your own <code>options.max_walltime</code> value. Second, you can use a
more relaxed convergence tolerance, that is, setting your own
<code>options.tolerance</code> value. Similar to MCMC, please note that
stopping VB iterations early may yield inaccurate inference results for
large-scale problems.</p>
<blockquote>
<p>Q: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30341297">Zhu and
Stephens (2018)</a> described a simple likelihood ratio calculation as a
sanity check for the more sophisticated enrichment analysis based on
RSS. Do you have software for this sanity check?</p>
</blockquote>
<p>A: Yes. I write a stand-alone script <a
href="https://github.com/stephenslab/rss/blob/master/misc/ash_lrt_31traits.R"><code>ash_lrt_31traits.R</code></a>
for this sanity check. Please carefully read the instruction in this
script. For more details of this sanity check, please see the caption of
<a
href="https://static-content.springer.com/esm/art%3A10.1038%2Fs41467-018-06805-x/MediaObjects/41467_2018_6805_MOESM1_ESM.pdf">Supplementary
Figure 17</a> in <a
href="https://www.ncbi.nlm.nih.gov/pubmed/30341297">Zhu and Stephens
(2018)</a>.</p>
<blockquote>
<p>Q: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30341297">Zhu and
Stephens (2018)</a> described an integrated framework based on RSS with
both enrichment and prioritization components, and we are only
interested in the gene prioritization component. How do we perform the
gene-level association test using RSS (something like Figure 3 in the
paper)?</p>
</blockquote>
<p>A: <a href="example_5a.html">Example 5 Part A</a> gives a
step-by-step illustration of gene-level association methods built on
RSS, with input and output files included. Since this example is based
on a relatively small simulation, it is highly recommended to run the
codes and see how the methods work.</p>
<blockquote>
<p>Q: Can I use the prioritization component of <a
href="https://www.ncbi.nlm.nih.gov/pubmed/30341297">Zhu and Stephens
(2018)</a> as a generic gene-level association testing method, assuming
that I do not have any gene set information available?</p>
</blockquote>
<p>A: Yes. When gene set information is not available, one can only fit
a baseline model (<span class="math inline">\(M_0:\theta=0\)</span>) and
obtain corresponding gene-level results. Indeed Figure 3 Panel A of <a
href="https://www.ncbi.nlm.nih.gov/pubmed/30341297">Zhu and Stephens
(2018)</a> illustrates a possible use scenario like this.</p>
</div>
<div id="genetics" class="section level2">
<h2>Genetics</h2>
<blockquote>
<p>Q: RSS uses the shrinkage estimator of LD in <a
href="https://www.ncbi.nlm.nih.gov/pubmed/21479081">Wen and Stephens
(2010)</a>, the computation of which requires the “scaled population
recombination rate” (<span class="math inline">\(\rho_{ij}\)</span>).
How do I compute <span class="math inline">\(\rho_{ij}\)</span>?</p>
</blockquote>
<p>A: Details of calculating <span
class="math inline">\(\rho_{ij}\)</span> are provided in these two great
papers: <a href="https://www.ncbi.nlm.nih.gov/pubmed/14704198">Li and
Stephens (2003)</a> and <a
href="https://www.ncbi.nlm.nih.gov/pubmed/21479081">Wen and Stephens
(2010)</a>. I also write a short <a
href="recombination.html">tutorial</a> to illustrate the calculation of
scaled population recombination rate (<span
class="math inline">\(\rho_{ij}\)</span>) using HapMap genetic map.</p>
<blockquote>
<p>Q: Most published RSS analyses used phased haplotypes to estimate LD.
Can I use unphased genotype data to compute this shrinkage estimator of
LD?</p>
</blockquote>
<p>A: Yes. You can feed <a
href="https://github.com/stephenslab/rss/blob/master/misc/get_corr.m"><code>get_corr.m</code></a>
an unphased genotype matrix and specify that <code>isgeno=true</code>.
There is a “0.5” correction term for genotype data, which comes from a
random mating assumption. Please see Section 2.4 of <a
href="https://www.ncbi.nlm.nih.gov/pubmed/21479081">Wen and Stephens
(2010)</a> for more details.</p>
<blockquote>
<p>Q: I know of <a
href="https://www.ncbi.nlm.nih.gov/pubmed/21479081">Wen and Stephens
(2010)</a>, but I’m not interested in imputation. Do you have software
that just computes the banded covariance or LD matrix?</p>
</blockquote>
<p>A: Yes. If you have MATLAB available, you can use <a
href="https://github.com/stephenslab/rss/blob/master/misc/get_corr.m"><code>get_corr.m</code></a>.
If you prefer R, you can use <a
href="https://github.com/stephenslab/rss/blob/master/misc/get_corr.R"><code>get_corr.R</code></a>.
I am also working on an R package <a
href="https://github.com/stephenslab/ldshrink"><code>ldshrink</code></a>
that can produce the banded LD matrix in my (very limited) spare
time.</p>
<blockquote>
<p>Q: In classical enrichment analyses, we often filter SNPs based on
their GWAS p-values (e.g., only using SNPs with p-values smaller than
<span class="math inline">\(5\times 10^{-8}\)</span> to test
enrichments). Can I apply the same SNP filter before running RSS methods
for enrichment analyses?</p>
</blockquote>
<p>A: No. The validity of enrichment analysis in RSS relies on an
accurately inferred ‘baseline’ model (<span
class="math inline">\(M_0\)</span>). Specifically, we need to learn from
the GWAS data about the average proportion of SNPs associated with a
trait without any enrichment information. To get a reasonable estimate
we need to consider genome-wide SNPs, including those with large
p-values. Here is a toy example to understand what might go wrong if we
only considered SNPs with small p-values. Suppose there are 1M
independent SNPs genome wide and only 100 of them are truly
trait-associated. The true proportion of trait-associated SNPs is
1e-4=100/1M. Suppose we identify 40 of 100 trait-associated SNPs with
GWAS p-value &lt;= 5e-8. If we only look at the filtered 1K SNPs with
GWAS p-value &lt;= 5e-8, then we get an overestimate 4e-2=40/1K.</p>
</div>
<div id="data" class="section level2">
<h2>Data</h2>
<blockquote>
<p>Q: If I want to use RSS methods, how should I prepare for the input
data?</p>
</blockquote>
<p>A: Thanks for your interest in trying RSS methods! To help you
prepare your own data for RSS-based analyses, I write <a
href="input_data.html">a short tutorial about input data formats</a>.
Please note that you need to store your input data in MAT-files if you
plan to use the MATLAB implementation of RSS methods.</p>
<blockquote>
<p>Q: Why the <code>SiRiS</code> matrix is coded as
<code>repmat((1./se),1,p) .* R .* repmat((1./se)',p,1)</code>, but not
<code>diag(1./se) .* R .* diag(1./se)</code>?</p>
</blockquote>
<p>A: The <code>SiRiS</code> matrix in our codes corresponds to <span
class="math inline">\({\bf S}^{-1}{\bf R}{\bf S}^{-1}\)</span> in our
math notations. We code this matrix as
<code>repmat((1./se),1,p) .* R .* repmat((1./se)',p,1)</code> instead of
the incorrect one <code>diag(1./se) .* R .* diag(1./se)</code> because
<a
href="https://www.mathworks.com/help/matlab/ref/times.html"><code>.*</code></a>
performs element-by-element multiplication in MATLAB.</p>
<blockquote>
<p>Q: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29399241">Zhu and
Stephens (2017)</a> reported a whole genome analysis of human adult
height data based on RSS. I am wondering whether you can share this
preprocessed height dataset.</p>
</blockquote>
<p>A: Yes. This preprocessed dataset is publicly available at <a
href="https://doi.org/10.5281/zenodo.1443565"
class="uri">https://doi.org/10.5281/zenodo.1443565</a>. and can be
referenced in a journal’s “Data availability” section as <a
href="https://doi.org/10.5281/zenodo.1443565"><img
src="https://zenodo.org/badge/DOI/10.5281/zenodo.1443565.svg"
alt="DOI" /></a>. For more information on this dataset, please see <a
href="height2014_data.html">this page</a>.</p>
<blockquote>
<p>Q: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30341297">Zhu and
Stephens (2018)</a> reported large-scale enrichment analyses of 4,026
gene sets across 31 complex human traits. Can you share these gene
sets?</p>
</blockquote>
<p>A: Yes. All 4,026 pre-processed gene sets used in this study
(including 3,913 biological pathways and 113 tissue-based gene sets) are
freely available at <a
href="https://github.com/xiangzhu/rss-gsea/tree/master/data"><code>xiangzhu/rss-gsea</code></a>.
These gene sets can be referenced in a journal’s “Data availability”
section as <a href="https://zenodo.org/badge/latestdoi/55633948"><img
src="https://zenodo.org/badge/55633948.svg" alt="DOI" /></a>. For more
information on these gene sets, please see <a
href="https://xiangzhu.github.io/rss-gsea/gene_set.html">this
page</a>.</p>
</div>
<div id="software" class="section level2">
<h2>Software</h2>
<blockquote>
<p>Q: RSS is cool, but it is implemented in MATLAB, which is a
proprietary programming language. Can RSS methods be made to work with
some open source language?</p>
</blockquote>
<p>A: To further improve the accessibility of RSS methods, I am
currently working on an R package <a
href="https://github.com/stephenslab/rssr"><code>rssr</code></a> in my
(very limited) spare time. Comments and suggestions are welcome!</p>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
