<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Xiang Zhu" />


<title>Function Reference</title>

<script src="site_libs/header-attrs-2.28/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>










<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">rss</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="setup.html">Setup</a>
</li>
<li>
  <a href="function.html">Functions</a>
</li>
<li>
  <a href="example.html">Examples</a>
</li>
<li>
  <a href="faq.html">FAQs</a>
</li>
<li>
  <a href="news.html">News</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephenslab/rss">
    <span class="fab fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Function Reference</h1>
<h4 class="author">Xiang Zhu</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-09-16
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>rss/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabrsstree941a146ebc0cf0ff1dfa5bbd4f2662140b4dddaetargetblank941a146a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/stephenslab/rss/tree/941a146ebc0cf0ff1dfa5bbd4f2662140b4dddae" target="_blank">941a146</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabrsstree941a146ebc0cf0ff1dfa5bbd4f2662140b4dddaetargetblank941a146a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/stephenslab/rss/tree/941a146ebc0cf0ff1dfa5bbd4f2662140b4dddae" target="_blank">941a146</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>rmd/function.Rmd</code>) and HTML
(<code>docs/function.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/rss/ca7cee47579836d0d29b46efebd6cd5930547d55/docs/function.html" target="_blank">ca7cee4</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2022-01-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/rss/blob/7f4dd4a660c41392c18f943ea04da25e5c6dc039/rmd/function.Rmd" target="_blank">7f4dd4a</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2022-01-24
</td>
<td>
wflow_publish("rmd/function.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/rss/bab3f582716fec43ec1d9285b4aff4484afb6075/docs/function.html" target="_blank">bab3f58</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2020-06-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/rss/843a7d9a0123fd73753576f4586e8181142398ed/docs/function.html" target="_blank">843a7d9</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2020-06-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/rss/blob/3b18fce8e46699d71472f399b083aeddda233add/rmd/function.Rmd" target="_blank">3b18fce</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2020-06-23
</td>
<td>
wflow_publish("rmd/function.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="markov-chain-monte-carlo-mcmc" class="section level2">
<h2>Markov chain Monte Carlo (MCMC)</h2>
<p>Details of MCMC algorithms for <a
href="https://github.com/stephenslab/rss/blob/master/src/rss_bvsr.m"><code>rss_bvsr.m</code></a>
and <a
href="https://github.com/stephenslab/rss/blob/master/src/rss_bslmm.m"><code>rss_bslmm.m</code></a>
are available in Supplementary Appendix B of <a
href="https://projecteuclid.org/euclid.aoas/1507168840">Zhu and Stephens
(2017)</a> Details of MCMC algorithms for <a
href="https://github.com/stephenslab/rss/blob/master/src/rss_ash.m"><code>rss_ash.m</code></a>
are available in this <a
href="https://www.stat.uchicago.edu/~xiangzhu/rss_mcmc.pdf">unpublished
note</a><sup>1</sup>. Note that only <a
href="https://github.com/stephenslab/rss/blob/master/src/rss_bvsr.m"><code>rss_bvsr.m</code></a>
and <a
href="https://github.com/stephenslab/rss/blob/master/src/rss_bslmm.m"><code>rss_bslmm.m</code></a>
were used to generate results in <a
href="https://projecteuclid.org/euclid.aoas/1507168840">Zhu and Stephens
(2017)</a>.</p>
<div id="rss_bvsr.m" class="section level3">
<h3><a
href="https://github.com/stephenslab/rss/blob/master/src/rss_bvsr.m"><code>rss_bvsr.m</code></a></h3>
<p>Fit the RSS-BVSR model that consists of RSS likelihood and BVSR prior
<a href="https://projecteuclid.org/euclid.aoas/1318514285">(Guan and
Stephens, 2011)</a>:</p>
<p><span class="math display">\[
\begin{aligned}
\widehat{\boldsymbol\beta} &amp;\sim {\cal N}({\bf
SRS}^{-1}{\boldsymbol\beta},{\bf SRS}),\\
\beta_j &amp;\sim \pi\cdot{\cal N}(0,\sigma_B^2) + (1-\pi)\cdot\delta_0,
\end{aligned}
\]</span></p>
<p>using a Metropolis-Hastings algorithm.</p>
</div>
<div id="rss_bslmm.m" class="section level3">
<h3><a
href="https://github.com/stephenslab/rss/blob/master/src/rss_bslmm.m"><code>rss_bslmm.m</code></a></h3>
<p>Fit the RSS-BSLMM model that consists of RSS likelihood and BSLMM
prior <a href="https://doi.org/10.1371/journal.pgen.1003264">(Zhou et
al, 2013)</a>:</p>
<p><span class="math display">\[
\begin{aligned}
\widehat{\boldsymbol\beta} &amp;\sim {\cal N}({\bf
SRS}^{-1}{\boldsymbol\beta},{\bf SRS}),\\
\beta_j &amp;\sim \pi\cdot{\cal N}(0,\sigma_B^2+\sigma_P^2) +
(1-\pi)\cdot{\cal N}(0,\sigma_P^2),
\end{aligned}
\]</span></p>
<p>using a component-wise MCMC algorithm.</p>
</div>
<div id="rss_ash.m" class="section level3">
<h3><a
href="https://github.com/stephenslab/rss/blob/master/src/rss_ash.m"><code>rss_ash.m</code></a></h3>
<p>Fit the RSS-ASH model that consists of RSS likelihood and ASH prior
<a href="https://doi.org/10.1093/biostatistics/kxw041">(Stephens,
2017)</a>:</p>
<p><span class="math display">\[
\begin{aligned}
\widehat{\boldsymbol\beta} &amp;\sim {\cal N}({\bf
SRS}^{-1}{\boldsymbol\beta},{\bf SRS}),\\
\beta_j &amp;\sim \pi_0 \cdot \delta_0 + {\textstyle\sum}_{k=1}^K \pi_k
\cdot {\cal N}(0,\sigma_k^2),
\end{aligned}
\]</span></p>
<p>using a component-wise MCMC algorithm.</p>
</div>
</div>
<div id="variational-bayes-vb" class="section level2">
<h2>Variational Bayes (VB)</h2>
<p>Details of VB algorithms, SQUAREM accelerator and parallel
implementation are available in Supplementary Notes of <a
href="https://www.nature.com/articles/s41467-018-06805-x">Zhu and
Stephens (2018)</a>. Note that only <a
href="https://github.com/stephenslab/rss/blob/master/src_vb/rss_varbvsr_squarem.m"><code>rss_varbvsr_squarem.m</code></a>
and <a
href="https://github.com/stephenslab/rss/blob/master/src_vb/rss_varbvsr_bigmem_squarem.m"><code>rss_varbvsr_bigmem_squarem.m</code></a>
were used to generate results in <a
href="https://www.nature.com/articles/s41467-018-06805-x">Zhu and
Stephens (2018)</a>. The other functions were developed merely for
testing and benchmarking.</p>
<div id="rss_varbvsr.m" class="section level3">
<h3><a
href="https://github.com/stephenslab/rss/blob/master/src_vb/rss_varbvsr.m"><code>rss_varbvsr.m</code></a></h3>
<p>Fit the following extended RSS-BVSR model</p>
<p><span class="math display">\[
\begin{aligned}
\widehat{\boldsymbol\beta} &amp;\sim {\cal N}({\bf
SRS}^{-1}{\boldsymbol\beta},{\bf SRS}),\\
\beta_j &amp;\sim \pi_j\cdot{\cal N}(0,\sigma_j^2) +
(1-\pi_j)\cdot\delta_0,
\end{aligned}
\]</span></p>
<p>using a mean-field VB algorithm. The VB algorithm largely follows <a
href="https://projecteuclid.org/euclid.ba/1339616726">Carbonetto and
Stephens (2012)</a>. This is an extended RSS-BVSR model because each SNP
<span class="math inline">\(j\)</span> can have its own hyper-parameters
<span class="math inline">\(\{\pi_j,\sigma_j^2\}\)</span>, whereas the
standard RSS-BVSR model assumes that all SNPs share the same
hyper-parameters <span
class="math inline">\(\{\pi,\sigma_B^2\}\)</span>.</p>
</div>
<div id="rss_varbvsr_squarem.m" class="section level3">
<h3><a
href="https://github.com/stephenslab/rss/blob/master/src_vb/rss_varbvsr_squarem.m"><code>rss_varbvsr_squarem.m</code></a></h3>
<p>This is a variant of <a
href="https://github.com/stephenslab/rss/blob/master/src_vb/rss_varbvsr.m"><code>rss_varbvsr.m</code></a>
with the SQUAREM accelerator <a
href="https://doi.org/10.1111/j.1467-9469.2007.00585.x">(Varadhan and
Roland, 2008)</a> added.</p>
</div>
<div id="rss_varbvsr_parallel.m" class="section level3">
<h3><a
href="https://github.com/stephenslab/rss/blob/master/src_vb/rss_varbvsr_parallel.m"><code>rss_varbvsr_parallel.m</code></a></h3>
<p>This is a parallel implementation of <a
href="https://github.com/stephenslab/rss/blob/master/src_vb/rss_varbvsr.m"><code>rss_varbvsr.m</code></a>
based on MATLAB <a
href="https://www.mathworks.com/help/distcomp/index.html">Parallel
Computing Toolbox</a>.</p>
</div>
<div id="rss_varbvsr_pasquarem.m" class="section level3">
<h3><a
href="https://github.com/stephenslab/rss/blob/master/src_vb/rss_varbvsr_pasquarem.m"><code>rss_varbvsr_pasquarem.m</code></a></h3>
<p>This is a parallel implementation of <a
href="https://github.com/stephenslab/rss/blob/master/src_vb/rss_varbvsr_squarem.m"><code>rss_varbvsr_squarem.m</code></a>
based on MATLAB <a
href="https://www.mathworks.com/help/distcomp/index.html">Parallel
Computing Toolbox</a>.</p>
</div>
<div id="rss_varbvsr_bigmem.m" class="section level3">
<h3><a
href="https://github.com/stephenslab/rss/blob/master/src_vb/rss_varbvsr_bigmem.m"><code>rss_varbvsr_bigmem.m</code></a></h3>
<p>This is a memory-efficient implementation of <a
href="https://github.com/stephenslab/rss/blob/master/src_vb/rss_varbvsr_parallel.m"><code>rss_varbvsr_parallel.m</code></a>.</p>
</div>
<div id="rss_varbvsr_bigmem_squarem.m" class="section level3">
<h3><a
href="https://github.com/stephenslab/rss/blob/master/src_vb/rss_varbvsr_bigmem_squarem.m"><code>rss_varbvsr_bigmem_squarem.m</code></a></h3>
<p>This is a memory-efficient implementation of <a
href="https://github.com/stephenslab/rss/blob/master/src_vb/rss_varbvsr_pasquarem.m"><code>rss_varbvsr_pasquarem.m</code></a>.</p>
</div>
</div>
<div id="miscellaneous" class="section level2">
<h2>Miscellaneous</h2>
<div id="import_1000g_vcf.sh" class="section level3">
<h3><a
href="https://github.com/stephenslab/rss/blob/master/misc/import_1000g_vcf.sh"><code>import_1000g_vcf.sh</code></a></h3>
<p>Output <a href="https://www.internationalgenome.org/data">1000
Genomes</a> phased haplotypes of a given list of SNPs in <a
href="https://mathgen.stats.ox.ac.uk/impute/impute_v2.html#input_options">IMPUTE
reference-panel format</a>.</p>
</div>
<div id="compute_pve.m" class="section level3">
<h3><a
href="https://github.com/stephenslab/rss/blob/master/src/compute_pve.m"><code>compute_pve.m</code></a></h3>
<p>Use GWAS summary data to estimate PVE (or SNP heritability), a
quantity defined by Equation 2.10 in <a
href="https://projecteuclid.org/euclid.aoas/1318514285">Guan and
Stephens (2011)</a>. This function corresponds to Equation 3.7 in <a
href="https://projecteuclid.org/euclid.aoas/1507168840">Zhu and Stephens
(2017)</a>.</p>
</div>
<div id="band_storage.m" class="section level3">
<h3><a
href="https://github.com/stephenslab/rss/blob/master/misc/band_storage.m"><code>band_storage.m</code></a></h3>
<p>Convert a symmetric, banded matrix to a compact matrix in such a way
that only the main diagonal and the nonzero super-diagonals are stored.
This function is used to reduce the file size of a large LD matrix.</p>
</div>
<div id="find_bandwidth.m" class="section level3">
<h3><a
href="https://github.com/stephenslab/rss/blob/master/misc/find_bandwidth.m"><code>find_bandwidth.m</code></a></h3>
<p>Find the bandwidth of a symmetric, banded matrix.</p>
</div>
<div id="get_corr.m" class="section level3">
<h3><a
href="https://github.com/stephenslab/rss/blob/master/misc/get_corr.m"><code>get_corr.m</code></a></h3>
<p>Compute linkage disequilibrium (LD) matrix using the shrinkage
estimator proposed in <a
href="https://www.ncbi.nlm.nih.gov/pubmed/21479081">Wen and Stephens
(2010)</a>. This function is also implemented in an <code>R</code>
package <a
href="https://github.com/stephenslab/ldshrink"><code>ldshrink</code></a>.</p>
</div>
<div id="data_maker.m" class="section level3">
<h3><a
href="https://github.com/stephenslab/rss/blob/master/misc/data_maker.m"><code>data_maker.m</code></a></h3>
<p>Simulate phenotype data from the genome-wide multiple-SNP model
described in <a href="https://doi.org/10.1371/journal.pgen.1003264">Zhou
et al (2013)</a>, and then compute the single-SNP summary statistics for
each SNP. This function was used in some simulation studies of <a
href="https://projecteuclid.org/euclid.aoas/1507168840">Zhu and Stephens
(2017)</a>.</p>
</div>
<div id="enrich_datamaker.m" class="section level3">
<h3><a
href="https://github.com/stephenslab/rss/blob/master/misc/enrich_datamaker.m"><code>enrich_datamaker.m</code></a></h3>
<p>Simulate phenotype data from the genetic association enrichment model
described in <a
href="https://doi.org/10.1371/journal.pgen.1003770">Carbonetto and
Stephens (2013)</a>, and then compute the single-SNP summary statistics
for each SNP. This function was used in some simulation studies of <a
href="https://www.nature.com/articles/s41467-018-06805-x">Zhu and
Stephens (2018)</a>.</p>
</div>
<div id="null_single.m-null_template.m" class="section level3">
<h3><a
href="https://github.com/stephenslab/rss/blob/master/src_vb/null_single.m"><code>null_single.m</code></a>
&amp; <a
href="https://github.com/stephenslab/rss/blob/master/src_vb/null_template.m"><code>null_template.m</code></a></h3>
<p>Fit genome-wide multiple-SNP “baseline model” to single-SNP summary
data, using <a
href="https://github.com/stephenslab/rss/tree/master/src_vb"><code>rss/src_vb</code>
functions</a>. These scripts were used in data analyses of <a
href="https://www.nature.com/articles/s41467-018-06805-x">Zhu and
Stephens (2018)</a>.</p>
</div>
<div id="gsea_wrapper.m-gsea_template.m" class="section level3">
<h3><a
href="https://github.com/stephenslab/rss/blob/master/src_vb/gsea_wrapper.m"><code>gsea_wrapper.m</code></a>
&amp; <a
href="https://github.com/stephenslab/rss/blob/master/src_vb/gsea_template.m"><code>gsea_template.m</code></a></h3>
<p>Fit genome-wide multiple-SNP “enrichment model” to single-SNP summary
data, using <a
href="https://github.com/stephenslab/rss/tree/master/src_vb"><code>rss/src_vb</code>
functions</a>. These scripts were used in data analyses of <a
href="https://www.nature.com/articles/s41467-018-06805-x">Zhu and
Stephens (2018)</a>.</p>
</div>
<div id="null_wrapper_fixsb.m-gsea_wrapper_fixsb.m"
class="section level3">
<h3><a
href="https://github.com/stephenslab/rss/blob/master/src_vb/null_wrapper_fixsb.m"><code>null_wrapper_fixsb.m</code></a>
&amp; <a
href="https://github.com/stephenslab/rss/blob/master/src_vb/gsea_wrapper_fixsb.m"><code>gsea_wrapper_fixsb.m</code></a></h3>
<p>Fit genome-wide multiple-SNP “baseline model” and “enrichment model”
to single-SNP summary data, using a fixed prior variance of causal
genetic effects (<span class="math inline">\(\sigma_B^2\)</span>) in <a
href="https://github.com/stephenslab/rss/tree/master/src_vb"><code>rss/src_vb</code>
functions</a>. These scripts were used in simulation studies of <a
href="https://www.nature.com/articles/s41467-018-06805-x">Zhu and
Stephens (2018)</a>.</p>
</div>
<div id="ash_lrt_31traits.r" class="section level3">
<h3><a
href="https://github.com/stephenslab/rss/blob/master/misc/ash_lrt_31traits.R"><code>ash_lrt_31traits.R</code></a></h3>
<p>Compute a simple likelihood ratio as a sanity check for the more
complicated enrichment analysis method developed in <a
href="https://www.nature.com/articles/s41467-018-06805-x">Zhu and
Stephens (2018)</a>. This likelihood ratio calculation is based on an
<code>R</code> package <a
href="https://cran.r-project.org/web/packages/ashr/index.html"><code>ashr</code></a>.</p>
<hr />
<p><strong>Footnotes:</strong></p>
<ol style="list-style-type: decimal">
<li>There is a missing multiplier term <span
class="math inline">\({\omega_k}/{\omega_1}\)</span> on the right-hand
side of Equation 5.1 in this note (page 4). However, this missing term
was correctly reflected in the source codes of <a
href="https://github.com/stephenslab/rss/blob/master/src/rss_ash.m"><code>rss_ash.m</code></a>,
in particular, Lines 80-81 of <a
href="https://github.com/stephenslab/rss/blob/master/src/update_bz.m"><code>update_bz.m</code></a>.
We thank Geyu Zhou <a href="mailto:geyu.zhou@yale.edu"
class="email">geyu.zhou@yale.edu</a> for pointing out this typo on
01/14/2019.</li>
</ol>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
