<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Xiang Zhu" />


<title>Progress Update</title>

<script src="site_libs/header-attrs-2.15/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>










<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">rss</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="setup.html">Setup</a>
</li>
<li>
  <a href="function.html">Functions</a>
</li>
<li>
  <a href="example.html">Examples</a>
</li>
<li>
  <a href="faq.html">FAQs</a>
</li>
<li>
  <a href="news.html">News</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephenslab/rss">
    <span class="fab fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Progress Update</h1>
<h4 class="author">Xiang Zhu</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-12-18
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>rss/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabrsstree8dbbf80e6c8a9db281f496f10e9cc32979e51d5atargetblank8dbbf80a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/stephenslab/rss/tree/8dbbf80e6c8a9db281f496f10e9cc32979e51d5a" target="_blank">8dbbf80</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabrsstree8dbbf80e6c8a9db281f496f10e9cc32979e51d5atargetblank8dbbf80a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/stephenslab/rss/tree/8dbbf80e6c8a9db281f496f10e9cc32979e51d5a" target="_blank">8dbbf80</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>rmd/news.Rmd</code>) and HTML
(<code>docs/news.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/rss/blob/8dbbf80e6c8a9db281f496f10e9cc32979e51d5a/rmd/news.Rmd" target="_blank">8dbbf80</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2022-12-18
</td>
<td>
wflow_publish("rmd/news.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/rss/762497e22fa35db6953bd373c69954603424fe82/docs/news.html" target="_blank">762497e</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2022-12-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/rss/blob/32cc296f67dd49d9ddd22018d981f35ad549e771/rmd/news.Rmd" target="_blank">32cc296</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2022-12-18
</td>
<td>
wflow_publish("rmd/news.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/rss/ee37b3365f7b78df9ac804cffa94a2811c996ed6/docs/news.html" target="_blank">ee37b33</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2022-12-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/rss/blob/9d679636b17c500cca04daab31a85c29e6d823d6/rmd/news.Rmd" target="_blank">9d67963</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2022-09-11
</td>
<td>
update news
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/rss/4da30281d7e3d633002ed195006567c321f5ef41/docs/news.html" target="_blank">4da3028</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2022-08-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/rss/blob/1616530cf233cfa1e457f688f6fe575f9949deb2/rmd/news.Rmd" target="_blank">1616530</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2022-08-31
</td>
<td>
wflow_publish("rmd/news.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/rss/a4c1991a840a6bf9dff0a0a2268939c513d292d6/docs/news.html" target="_blank">a4c1991</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2022-08-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/rss/blob/1116f64b69601c5cf1097f0c4e822224f77ab958/rmd/news.Rmd" target="_blank">1116f64</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2022-08-29
</td>
<td>
wflow_publish("rmd/news.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/rss/bc547cf20a1adf6e850b706412c9caa8bf7806a2/docs/news.html" target="_blank">bc547cf</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2022-08-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/rss/blob/6cba980434ee021180f058dbf47b3d6047af57e7/rmd/news.Rmd" target="_blank">6cba980</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2022-08-29
</td>
<td>
update news
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/rss/blob/6c18e124ee502e1568b638fa7e379751441403f8/rmd/news.Rmd" target="_blank">6c18e12</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2022-05-05
</td>
<td>
update news
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/rss/blob/4696557222df9ab077c5bc3870888454a4b68c77/rmd/news.Rmd" target="_blank">4696557</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2022-03-31
</td>
<td>
rename nhs to pcb
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/rss/8d749ff3f02ec68bca37b6ce505e5dbe31e148bb/docs/news.html" target="_blank">8d749ff</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2022-01-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/rss/blob/f3d289f3520fa94e0b3c13606d3948e16e5e9491/rmd/news.Rmd" target="_blank">f3d289f</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2022-01-24
</td>
<td>
wflow_publish("rmd/news.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/rss/412881d19927d6779aaf83128def3bef2e886d78/docs/news.html" target="_blank">412881d</a>
</td>
<td>
xiangzhu
</td>
<td>
2021-06-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/rss/blob/d4400e925895b2165e01f3fbd6310dd1ee556721/rmd/news.Rmd" target="_blank">d4400e9</a>
</td>
<td>
xiangzhu
</td>
<td>
2021-06-07
</td>
<td>
wflow_publish("rmd/news.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/rss/4c4b855728413786b8b1a7c62897f4e5b8eb22c2/docs/news.html" target="_blank">4c4b855</a>
</td>
<td>
xiangzhu
</td>
<td>
2021-06-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/rss/blob/73865eca1331f0880593cf0016d25866db24b7b5/rmd/news.Rmd" target="_blank">73865ec</a>
</td>
<td>
xiangzhu
</td>
<td>
2021-06-07
</td>
<td>
wflow_publish("rmd/news.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/rss/8895c9d8e2c08a9c7ff604592d7226c1da7b29bb/docs/news.html" target="_blank">8895c9d</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2021-03-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/rss/blob/293645c93ab04499e6972da3b99450d2b488d85a/rmd/news.Rmd" target="_blank">293645c</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2021-03-04
</td>
<td>
wflow_publish("rmd/news.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/rss/cc8973465c0ed270c9d6226d173e4e80a936503b/docs/news.html" target="_blank">cc89734</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2020-06-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/rss/blob/97b9b7fb6a0bb2b9a7ec6600222e9833d2438cb4/rmd/news.Rmd" target="_blank">97b9b7f</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2020-06-24
</td>
<td>
wflow_publish("rmd/news.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/rss/bab3f582716fec43ec1d9285b4aff4484afb6075/docs/news.html" target="_blank">bab3f58</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2020-06-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/rss/d8bfc6c16d5873cdf9d28e8197d80b3be98bb6e0/docs/news.html" target="_blank">d8bfc6c</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2020-06-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/rss/blob/6485a10b97b888556d48a0b8f31931181a7265c2/rmd/news.Rmd" target="_blank">6485a10</a>
</td>
<td>
Xiang Zhu
</td>
<td>
2020-06-23
</td>
<td>
wflow_publish("rmd/news.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<ul>
<li><p>12/18/2022: Xiang presented <a
href="https://doi.org/10.1038/s41467-021-22588-0">RSS-NET</a> and its
extensions to analyze <a
href="https://doi.org/10.1073/pnas.2115601119">H3K27ac HiChIP
networks</a> and <a
href="https://doi.org/10.1101/2022.08.19.504589">sequence-conserved
enhancer-like elements</a> at the 2022 CMStatistics, London,
UK.</p></li>
<li><p>09/14/2022: Xiang presented the <a
href="https://doi.org/10.1214/17-aoas1046">RSS likelihood</a> and its
various applications in human genetics at the Department of Statistics,
University of Nebraska-Lincoln, Lincoln, NE.</p></li>
<li><p>08/19/2022: Xiang extended <a
href="https://doi.org/10.1038/s41467-021-22588-0">RSS-NET</a> to analyze
<a href="https://github.com/SUwonglab/m2h-aele">sequence-conserved
enhancer-like elements</a>. This <a
href="https://doi.org/10.1101/2022.08.19.504589">extension</a> was
posted on <em>bioRxiv</em>.</p></li>
<li><p>08/04/2022: Xiang applied <a
href="https://doi.org/10.1038/s41467-021-22588-0">RSS-NET</a> to
identify relevant tissues and cell types for five lipid traits based on
the latest <a href="https://doi.org/10.1038/s41586-021-04064-3">GWAS
results from GLGC</a>. This <a
href="https://doi.org/10.1016/j.ajhg.2022.06.012">application</a> was
published in <em>American Journal of Human Genetics</em>.</p></li>
<li><p>08/01/2022: Xiang applied <a
href="https://doi.org/10.1038/s41467-018-06805-x">RSS-E</a> to
prioritize pathways and genes for coronary artery disease based on the
latest <a href="https://doi.org/10.1038/s41591-022-01891-3">GWAS results
from MVP</a>. This <a
href="https://doi.org/10.1038/s41591-022-01891-3">application</a> was
published in <em>Nature Medicine</em>.</p></li>
<li><p>05/18/2022: Xiang received the <a
href="https://csua.ssri.psu.edu/">Penn State Consortium on Substance Use
and Addiction</a> Seed Grant as the Lead PI to incorporate single-cell
genomics and multi-ancestry GWAS into the <a
href="https://doi.org/10.1214/17-aoas1046">RSS likelihood</a> framework,
with a special focus on genetics of substance use traits in diverse
human populations.</p></li>
<li><p>05/02/2022: Xiang presented <a
href="https://doi.org/10.1038/s41467-021-22588-0">RSS-NET</a> at the
Department of Statistics, University of Missouri, Columbia, MO.</p></li>
<li><p>03/30/2022: Xiang received the <a
href="https://www.icds.psu.edu/">Penn State Institute for Computational
and Data Sciences</a> Seed Grant as the Lead PI to incorporate
single-cell genomics and multi-ancestry GWAS into the <a
href="https://doi.org/10.1214/17-aoas1046">RSS likelihood</a> framework,
with a special focus on genetics of cardiometabolic diseases in diverse
human populations. <em>Penn State Today</em> published a <a
href="https://www.psu.edu/news/institute-computational-and-data-sciences/story/seed-grants-fund-projects-tackle-huge-scientific/">press
release</a>.</p></li>
<li><p>01/04/2022: Xiang extended <a
href="https://doi.org/10.1038/s41467-021-22588-0">RSS-NET</a> to analyze
regulatory networks derived from H3K27ac HiChIP experiments. This <a
href="https://doi.org/10.1073/pnas.2115601119">extension</a> was
published in <em>Proceedings of the National Academy of
Sciences</em>.</p></li>
<li><p>06/29/2021: Xiang presented <a
href="https://doi.org/10.1038/s41467-021-22588-0">RSS-NET</a> at the
2021 Virtual ISBA World Meeting.</p></li>
<li><p>05/14/2021: <a
href="https://doi.org/10.1038/s41467-021-22588-0">RSS-NET</a> was
published in <em>Nature Communications</em>.</p></li>
<li><p>05/07/2021: Xiang presented <a
href="https://doi.org/10.1038/s41467-021-22588-0">RSS-NET</a> at the
Program in Computational Biology Seminar, University of Chicago,
Chicago, IL.</p></li>
<li><p>04/15/2021: Xiang presented <a
href="https://doi.org/10.1038/s41467-021-22588-0">RSS-NET</a> at the
Division of Biostatistics and Bioinformatics, Department of Public
Health Sciences, Penn State College of Medicine, Hershey, PA.</p></li>
<li><p>04/14/2021: Xiang presented <a
href="https://doi.org/10.1038/s41467-021-22588-0">RSS-NET</a> at the
2021 Virtual CSHL Probabilistic Modeling in Genomics Meeting.</p></li>
<li><p>03/03/2021: <a
href="https://doi.org/10.1038/s41467-021-22588-0">RSS-NET</a> was
accepted by <em>Nature Communications</em>.</p></li>
<li><p>11/04/2020: Xiang presented <a
href="https://doi.org/10.1038/s41467-021-22588-0">RSS-NET</a> at the
2020 Virtual CSHL Biological Data Science Meeting.</p></li>
<li><p>08/05/2020: Xiang presented <a
href="https://doi.org/10.1038/s41467-021-22588-0">RSS-NET</a> at the
2020 Virtual JSM.</p></li>
<li><p>04/28/2020: Xiang presented <a
href="https://doi.org/10.1038/s41467-021-22588-0">RSS-NET</a> at the
Department of Statistics, Stanford University, Stanford, CA.</p></li>
<li><p>03/14/2020: Xiang posted <a
href="https://doi.org/10.1038/s41467-021-22588-0">RSS-NET</a> on
<em>bioRxiv</em>.</p></li>
<li><p>02/10/2020: Xiang presented <a
href="https://doi.org/10.1038/s41467-021-22588-0">RSS-NET</a> at the
Department of Biostatistics and Bioinformatics, Rollins School of Public
Health, Emory University, Atlanta, GA.</p></li>
<li><p>01/23/2020: Xiang presented <a
href="https://doi.org/10.1038/s41467-021-22588-0">RSS-NET</a> at the
Department of Statistics and Huck Institutes of the Life Sciences,
Pennsylvania State University, University Park, PA.</p></li>
<li><p>01/15/2020: Xiang presented <a
href="https://doi.org/10.1038/s41467-021-22588-0">RSS-NET</a> at the
Department of Statistics, Wharton School, University of Pennsylvania,
Philadelphia, PA.</p></li>
<li><p>03/27/2019: <a
href="https://doi.org/10.1038/s41467-018-06805-x">RSS-E</a> was
recommended in <a
href="https://doi.org/10.3410/f.734261365.793558023">F1000Prime</a> by
F1000 Faculty Member <a
href="https://f1000.com/prime/thefaculty/member/499999771097550974">Jason
Flannick</a>.</p></li>
<li><p>01/10/2019: The Research Computing Center at University of
Chicago wrote an <a
href="https://rcc.uchicago.edu/about-rcc/news-features/new-statistical-method-gives-second-life-gwas-data">article</a>
on <a href="https://doi.org/10.1214/17-aoas1046">RSS likelihood</a> and
<a href="https://doi.org/10.1038/s41467-018-06805-x">RSS-E</a>.</p></li>
<li><p>10/19/2018: <a
href="https://doi.org/10.1038/s41467-018-06805-x">RSS-E</a> was
published in <em>Nature Communications</em>.</p></li>
<li><p>10/09/2018: Xiang presented <a
href="https://doi.org/10.1038/s41467-018-06805-x">RSS-E</a> at the
MVP/ERIC/Assimes Group Meeting, Stanford University, Stanford,
CA.</p></li>
<li><p>09/29/2018: <a
href="https://doi.org/10.1038/s41467-018-06805-x">RSS-E</a> was accepted
by <em>Nature Communications</em>.</p></li>
<li><p>10/19/2017: Xiang presented <a
href="https://doi.org/10.1038/s41467-018-06805-x">RSS-E</a> at the 2017
ASHG, Orlando, FL.</p></li>
<li><p>10/05/2017: <a href="https://doi.org/10.1214/17-aoas1046">RSS
likelihood</a> was published in <em>Annals of Applied
Statistics</em>.</p></li>
<li><p>08/03/2017: Xiang presented <a
href="https://doi.org/10.1038/s41467-018-06805-x">RSS-E</a> at the 2017
JSM, Baltimore, MD. This work won an <a
href="https://stat.uchicago.edu/academics/graduate-programs/graduate-student-resources/academic-life/student-awards/#collapse13">American
Statistical Association Student Paper Award</a>.</p></li>
<li><p>07/07/2017: Xiang posted <a
href="https://doi.org/10.1038/s41467-018-06805-x">RSS-E</a> on
<em>bioRxiv</em>.</p></li>
<li><p>06/30/2017: Xiang presented <a
href="https://doi.org/10.1214/17-aoas1046">RSS likelihood</a> and <a
href="https://doi.org/10.1038/s41467-018-06805-x">RSS-E</a> at
Dissertation Presentation and Defense, Department of Statistics,
University of Chicago, Chicago, IL.</p></li>
<li><p>06/28/2017: Xiang presented <a
href="https://doi.org/10.1038/s41467-018-06805-x">RSS-E</a> at the 2017
ICSA Applied Statistics Symposium, Chicago, IL.</p></li>
<li><p>05/02/2017: Xiang presented <a
href="https://doi.org/10.1038/s41467-018-06805-x">RSS-E</a> at the Mind
Bytes 2017, University of Chicago, Chicago, IL. This work won the <a
href="https://mindbytes.uchicago.edu/2017/callforposters.php#tab8">Mind
Bytes Award for Big-Data Research</a>.</p></li>
<li><p>04/04/2017: <a href="https://doi.org/10.1214/17-aoas1046">RSS
likelihood</a> was accepted by <em>Annals of Applied
Statistics</em>.</p></li>
<li><p>02/22/2017: Xiang presented <a
href="https://doi.org/10.1214/17-aoas1046">RSS likelihood</a> at the
Department of Statistics, Stanford University, Stanford, CA.</p></li>
<li><p>01/11/2017: Xiang presented <a
href="https://doi.org/10.1038/s41467-018-06805-x">RSS-E</a> at the
Department of Human Genetics, University of Chicago, Chicago,
IL.</p></li>
<li><p>12/02/2016: Xiang presented <a
href="https://doi.org/10.1214/17-aoas1046">RSS likelihood</a> as part of
a GWAS tutorial at the Program in Computational Biology Seminar,
University of Chicago, Chicago, IL.</p></li>
<li><p>10/20/2016: Xiang presented <a
href="https://doi.org/10.1038/s41467-018-06805-x">RSS-E</a> at the 2016
ASHG, Vancouver, BC, Canada.</p></li>
<li><p>08/23/2016: <a href="https://pa.msu.edu/profile/hsu/">Stephen
Hsu</a> wrote a <a
href="https://infoproc.blogspot.com/2016/08/bayesian-large-scale-multiple.html">blog</a>
about <a href="https://doi.org/10.1214/17-aoas1046">RSS
likelihood</a>.</p></li>
<li><p>07/31/2016: Xiang presented <a
href="https://doi.org/10.1214/17-aoas1046">RSS likelihood</a> at the
2016 JSM, Chicago, IL.</p></li>
<li><p>03/04/2016: Xiang posted <a
href="https://doi.org/10.1214/17-aoas1046">RSS likelihood</a> on
<em>bioRxiv</em>.</p></li>
<li><p>03/04/2016: Xiang presented <a
href="https://doi.org/10.1038/s41467-018-06805-x">RSS-E</a> at the
Program in Computational Biology Seminar, University of Chicago,
Chicago, IL.</p></li>
<li><p>11/11/2015: Xiang presented <a
href="https://doi.org/10.1214/17-aoas1046">RSS likelihood</a> at the
Dissertation Proposal Presentation, Department of Statistics, University
of Chicago, Chicago, IL.</p></li>
<li><p>10/27/2015: Xiang presented <a
href="https://doi.org/10.1038/s41467-018-06805-x">RSS-E</a> at the Mind
Bytes 2015, University of Chicago, Chicago, IL.</p></li>
<li><p>10/15/2015: Xiang presented <a
href="https://doi.org/10.1038/s41467-018-06805-x">RSS-E</a> at the 2015
ProbGen, Cold Spring Harbor Laboratory, Cold Spring Harbor, NY.</p></li>
<li><p>10/09/2015: Xiang presented <a
href="https://doi.org/10.1214/17-aoas1046">RSS likelihood</a> at the
2015 ASHG, Baltimore, MD.</p></li>
<li><p>05/05/2015: Xiang received the inaugural <a
href="https://stat.uchicago.edu/academics/graduate-programs/graduate-student-resources/academic-life/student-awards/#collapse22">David
Wallace Award for Applied Statistics</a> for the work on <a
href="https://doi.org/10.1214/17-aoas1046">RSS likelihood</a>.</p></li>
<li><p>02/27/2015: Xiang presented <a
href="https://doi.org/10.1214/17-aoas1046">RSS likelihood</a> at the
Program in Computational Biology Seminar, University of Chicago,
Chicago, IL.</p></li>
</ul>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
